### PR TITLE
Migrate to Bootstrap 4

### PR DESCRIPTION
**Pull request by jayhesselberth**
Current preview at: https://awesome-tesla-bf9106.netlify.com/

@hadley There are a few remaining TODO items but this is ready for review.

I think we should do a release and then merge this. Templates especially will require a few tweaks to accommodate BS4.

### CODE REVIEW

**Comment by jayhesselberth** [comment]
This still needs a good amount of style tweaking but most functionality is there (including sticky sidebar).

### Major changes

* Tables of contents in sidebars now use [bootstrap-toc](https://afeld.github.io/bootstrap-toc/). TOC styling needs to better adopt colors from bootswatch.
* New card layout for reference index. The bootstrap-toc improves multi-level display and navigation for large function references.
* The navbar HTML structure changed considerably between BS3 and 4 and so no longer relies on `rmarkdown::render_navbar_links`.

### Breaking changes

* Some bootswatch themes disappeared between v3 and v4. Notably the tidytemplate default (paper) is not in bootswatch v4; closest theme might be "journal"

**Comment by hadley** [comment]
Sounds good. I’ll plan to start the release process next week (currently in England)

File: inst/templates/content-reference-topic.html [hunk 1]
```diff
@@ -19,16 +19,13 @@ <h1>{{{title}}}</h1>
...
-    <table class="ref-arguments">
-    <colgroup><col class="name" /><col class="desc" /></colgroup>
+    <h2>Arguments</h2>
+    <dl class="row" id="ref-arguments">
     {{#arguments}}
-    <tr>
-      <th>{{{name}}}</th>
-      <td>{{{description}}}</td>
-    </tr>
+      <dt class="col-sm-3 text-truncate" id="ref-arguments name">{{{name}}}</dt>
```
Comment (slowkow) [inline inst/templates/content-reference-topic.html]: Could I please ask you to delete `text-truncate`?

On my site, this class prevents me from reading the documentation for the functions. Is there a good reason for it?

File: inst/templates/content-reference-index.html [hunk 2]
```diff
@@ -1,41 +1,33 @@
...
 
-    <table class="ref-index{{#has_icons}} ref-index-icons{{/has_icons}}">
-
-    <colgroup>
-      {{#has_icons}}<col class="icon" />{{/has_icons}}
-      <col class="alias" />
-      <col class="title" />
-    </colgroup>
+    {{#sections}}
+    <div class="card">
```
Comment (slowkow) [inline inst/templates/content-reference-index.html]: Could we add class `mb-2` to the card, so the cards are not touching?

**Comment by jayhesselberth** [comment]
@slowkow I made those changes. This is all experimental so if you have other tweaks let me know.

**Comment by jayhesselberth** [comment]
This might the way to go: https://github.com/rstudio/rmarkdown/pull/1706

**Comment by jayhesselberth** [comment]
Closing for now in favor of a likely better strategy once rstudio/rmarkdown#1706 is merged.

**Pull request closed without merging**